diff -Nur make-4.2.1.orig/main.c make-4.2.1/main.c
--- make-4.2.1.orig/main.c	2019-09-16 17:48:01.919845489 -0700
+++ make-4.2.1/main.c	2019-09-16 17:49:05.890104766 -0700
@@ -44,6 +44,9 @@
 #ifdef HAVE_FCNTL_H
 # include <fcntl.h>
 #endif
+#ifdef HAVE_UNISTD_H
+# include <unistd.h>
+#endif
 
 #ifdef _AMIGA
 int __stack = 20000; /* Make sure we have 20K of stack space */
@@ -328,6 +331,9 @@
 
 char cmd_prefix = '\t';
 
+/* The value make will nice itself to. */
+static int arg_nice = 0;
+
 
 /* The usage output.  We write it this way to make life easier for the
    translators, especially those trying to translate to right-to-left
@@ -384,6 +390,10 @@
   -p, --print-data-base       Print make's internal database.\n"),
     N_("\
   -q, --question              Run no recipe; exit status says if up to date.\n"),
+#ifdef HAVE_UNISTD_H
+    N_("\
+  --nice                      Set a nice value.\n"),
+#endif
     N_("\
   -r, --no-builtin-rules      Disable the built-in implicit rules.\n"),
     N_("\
@@ -471,6 +481,7 @@
       "warn-undefined-variables" },
     { CHAR_MAX+6, strlist, &eval_strings, 1, 0, 0, 0, 0, "eval" },
     { CHAR_MAX+7, string, &sync_mutex, 1, 1, 0, 0, 0, "sync-mutex" },
+    { CHAR_MAX+8, positive_int, &arg_nice, 1, 0, 0, 0, 0, "nice" },
     { 0, 0, 0, 0, 0, 0, 0, 0, 0 }
   };
 
@@ -1494,6 +1505,15 @@
       arg_job_slots = env_slots;
   }
 
+  /* Nice ourselves, if requested. */
+  if (arg_nice)
+    {
+#ifdef HAVE_UNISTD_H
+      if (nice(0) == nice(arg_nice))
+#endif
+        O (error, NILF, _("warning: unable to nice"));
+    }
+
   /* Set a variable specifying whether stdout/stdin is hooked to a TTY.  */
 #ifdef HAVE_ISATTY
   if (isatty (fileno (stdout)))
